File: nas_spec/code_samples/Node/sessions@{id}@collect-data/put.js
import { Checkout } from 'checkout-node-sdk';

let cko = new Checkout('your_client_secret_here', {
	client: 'ack_XXXXXXXXXXXX',
	scope: ['sessions:browser'], // array of scopes
	environment: 'sandbox', // or "production"
});

// Or if you use api keys:
// const cko = new Checkout('sk_sbox_XXX', { pk: 'pk_sbox_XXX'}});

try {
	let session = await cko.sessions.update('sid_rwhwl4kb3eeenglibbvej2qtdy', {
		channel: 'browser',
		accept_header: 'Accept:  *.*, q=0.1',
		java_enabled: true,
		language: 'FR-fr',
		color_depth: '16',
		screen_height: '1080',
		screen_width: '1920',
		timezone: '60',
		user_agent:
			'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36',
		ip_address: '1.12.123.255',
	});
} catch (err) {
	console.log(err.name);
}

File: nas_spec/code_samples/Go/sessions@{id}@collect-data/put.go
// For more information please refer to https://github.com/checkout/checkout-sdk-go
import (
	"github.com/checkout/checkout-sdk-go"
	"github.com/checkout/checkout-sdk-go/common"
	"github.com/checkout/checkout-sdk-go/configuration"
	"github.com/checkout/checkout-sdk-go/sessions/channels"
)

// API Keys
api, err := checkout.
			Builder().
			StaticKeys().
			WithSecretKey("secret_key").
			WithEnvironment(configuration.Sandbox()). // or Environment.PRODUCTION
			Build()
if err != nil {
	return nil, err
}

// OAuth
api, err := checkout.
			Builder().
			OAuth().
			WithClientCredentials("client_id", "client_secret").
			WithScopes([]string{configuration.SessionsApp, configuration.SessionsBrowser}).
			WithEnvironment(configuration.Sandbox()). // or Environment.PRODUCTION
			Build()
if err != nil {
	return nil, err
}

channel := channels.NewBrowserSession()
channel.AcceptHeader = "Accept:  *.*, q=0.1"
channel.JavaEnabled = true
channel.Language = "FR-fr"
channel.ColorDepth = "16"
channel.ScreenWidth = "1920"
channel.ScreenHeight = "1080"
channel.Timezone = "60"
channel.UserAgent = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
channel.ThreeDsMethodCompletion = common.Y
channel.IpAddress = "1.12.123.255"

response, err := api.Sessions.UpdateSession("session_id", channel,"session_secret") // "session_secret" is optional
if err != nil {
	return nil, err
}

return response, nil
